test/integration.py

```diff
@@ -41,24 +41,24 @@ def setUp(self):
         self.oid_source_dir = os.path.dirname(
             os.path.dirname(os.path.abspath(__file__))
         )
+        # Assume we started in the CMake build directory
+        self.build_dir = self.original_working_directory
 
         # This needs to be a class variable, otherwise it won't be referenced
         # by any object alive by the end of this class method's execution and
         # and the directory will be automatically removed before executing the
         # tests themselves.
-        self.temp = tempfile.TemporaryDirectory(
-            dir=os.path.join(self.oid_source_dir, "build/")
-        )
+        self.temp = tempfile.TemporaryDirectory(dir=self.build_dir)
         os.chdir(self.temp.name)
 
-        self.oid = os.path.join(self.oid_source_dir, "build/oid")
-        self.oid_conf = os.path.join(self.oid_source_dir, "build/testing.oid.toml")
+        self.oid = os.getenv("OID")
+        self.oid_conf = os.path.join(self.build_dir, "..", "testing.oid.toml")
```

Comment:
I left this one as is because it was already doing it and this file is only put there by the `Makefile` anyway. We should probably fix that but I see it as a separate problem.